test/util/util.go

```diff
@@ -291,3 +291,18 @@ func ExpectResourceFlavorToBeDeleted(ctx context.Context, k8sClient client.Clien
 		return k8sClient.Get(ctx, client.ObjectKeyFromObject(rf), &newRF)
 	}, Timeout, Interval).Should(testing.BeNotFoundError())
 }
+
+func SetWorkloadAdmission(wl *kueue.Workload, admission *kueue.Admission) bool {
+	wl.Status.Admission = admission
```

Comment:
maybe `if admission == nil { panic }`

we don't really want to use this function to evict/preempt. Or if we do, we should reset the condition to False.

Also we could make this helper do more: call the client Update.

Reply:
done